server/handler.go

```diff
@@ -525,11 +482,11 @@ func (h *Handler) pollForClosedConnection(nc conntainer, errChan chan error, qui
 		st, err := sockstate.GetInodeSockState(t.Port, inode)
 		switch st {
 		case sockstate.Broken:
-			logrus.Tracef("socket state is broken, returning error")
+			logrus.Warn("socket state is broken, returning error")
 			errChan <- ErrConnectionWasClosed.New()
 			return
 		case sockstate.Error:
-			errChan <- err
+			logrus.Warnf("Connection checker exiting, got err checking sockstate: %v", err)
```

Comment:
We're changing the behavior here as well to just warn and shutdown the checking goroutine. This functionality is nice-to-have, but by no means critical.